From f0522a8b6360fbd5dcc3b23ea2e4aa77f7c6eba9 Mon Sep 17 00:00:00 2001
From: Ravi B <ravibabu@ti.com>
Date: Tue, 22 Mar 2011 13:56:24 +0530
Subject: [PATCH 0528/1092] usb-musb: Fix for modular support issue for TI81XX
 (PIO Mode)

This patch fixes the modular support issue for dual
musb instances in ti81xx (for pio mode).

Signed-off-by: Ravi B <ravibabu@ti.com>
Signed-off-by: Sriramakrishnan A G <srk@ti.com>
Signed-off-by: Vaibhav Hiremath <hvaibhav@ti.com>
---
 drivers/usb/musb/ti81xx.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/musb/ti81xx.c b/drivers/usb/musb/ti81xx.c
index ac76247..be2e5a2 100644
--- a/drivers/usb/musb/ti81xx.c
+++ b/drivers/usb/musb/ti81xx.c
@@ -1108,13 +1108,13 @@ static int __init ti81xx_probe(struct platform_device *pdev)
 	int                             ret;
 	struct clk *otg_fck = NULL;
 
+	pdev->num_resources = 2;
 	glue = kzalloc(sizeof(*glue), GFP_KERNEL);
 	if (!glue) {
 		dev_err(&pdev->dev, "unable to allocate memory\n");
 		return -ENOMEM;
 	}
 
-	dev_set_name(&pdev->dev, "musb-ti81xx");
 	if (cpu_is_ti816x())
 		glue->clk = clk_get(&pdev->dev, "usbotg_ick");
 	else
@@ -1211,6 +1211,7 @@ static int __exit ti81xx_remove(struct platform_device *pdev)
 	clk_put(glue->clk);
 	clk_put(glue->phy_clk);
 	kfree(glue);
+	pdev->num_resources = 0;
 
 	return 0;
 }
-- 
1.7.11.2

